compat/xis.h

```diff
@@ -9,11 +9,8 @@
 #ifndef _SQUID_COMPAT_XIS_H
 #define _SQUID_COMPAT_XIS_H
 
-#if HAVE_CTYPE_H
-#include <ctype.h>
-#endif
-
 #if __cplusplus
```

Comment:
The removal of the above lines does not look right to me. This file provides wrappers to standard isfoo() functions. It should include the header that declares those functions because all legitimate xis.h users need those declarations. This change goes against the header inclusion principles that we should formalize and adopt IMO.

It looks like the compiler will force you to fix this, but I wanted to add this comment to emphasize that we should include that standard header (once in each `#if` clause below) not (just) because the compiler forces us to do that.